### DISCUSSION TITLE
diagonal concatenation of list of matrices in python

### DISCUSSION THREAD

**Discussion by sandeep026** [casadi/casadi#3406] [Q&A] [answered]
I have a list of matrices and I want to diagonally concatenate them.

Presently, I observed that diagcat( ) takes arguments in the following form.

```
A1=DM_rand(5,5)
A2=DM_rand(5,5)
A3=DM_rand(5,5)
diagcat(A1,A2,A3)
```

I have a list containing matrices [A1,A2,A3]. In real life scenario the number of matrices can be varying, so I wrote the following script
to automate the task.

```
B=[A1,A2,A3]

for i in range(len(B)):
    
    if i==0:
        C=B[i]
    else:    
        C=cs.diagcat(C,B[i])

print(C)   
```

There are 2 things I need help with.

1. Can this be made more efficent? Like parallelization using map()?
2.  Does diagcat support diagcat(B) i.e. on the list of matrices? I tried but it didn't work.  

Thankful for your suggestions

**Comment by jgillis** [comment, marked as answer]
`cs.diagcat(*B)` is far more efficient

**Reply by sandeep026** [reply]
Thanks for the quick help @jgillis . I switched from MATLAB to python recently. Just learned about passing variable number of arguments  (*args) now.
